lib/matplotlib/axes/_axes.py

```diff
@@ -4792,6 +4792,8 @@ def _pcolorargs(funcname, *args, **kw):
 
         if len(args) == 3:
             X, Y, C = args
+            X = np.asarray(X)
+            Y = np.asarray(Y)
```

Comment:
From the pcolormesh docstring: `*C* may be a masked array, but *X* and *Y* may not.`